packages/ui-menu/src/Menu/MenuItem/styles.js

```diff
@@ -0,0 +1,124 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/**
+ * Generates the style object from the theme and provided additional information
+ * @param  {Object} componentTheme The theme variable object.
+ * @param  {Object} props the props of the component, the style is applied to
+ * @param  {Object} state the state of the component, the style is applied to
+ * @return {Object} The final style object, which will be used in the component
+ */
+const generateStyle = (componentTheme, props, state) => {
+  const { type, disabled } = props
+  const { role } = state
+
+  const isMenuitemradioOrMenuitemcheckbox =
+    type === 'checkbox' || type === 'radio'
+
+  const flyoutIconStyles =
+    type === 'flyout'
+      ? {
+          insetInlineStart: 'auto',
+          insetInlineEnd: componentTheme.iconPadding
+        }
+      : {}
+
+  const roleStyles = isMenuitemradioOrMenuitemcheckbox
+    ? {
+        paddingInlineStart: componentTheme.labelPadding
+      }
+    : {}
+  const roleIconStyles =
+    role === 'menuitemradio' || role === 'menuitemcheckbox'
+      ? {
+          insetInlineStart: componentTheme.iconPadding,
+          insetInlineEnd: 'auto'
+        }
+      : {}
+  const disabledStyles = disabled
+    ? {
+        cursor: 'not-allowed',
+        pointerEvents: 'none',
+        opacity: 0.5
+      }
+    : {}
+  return {
+    menuItem: {
+      label: 'menuItem',
+      position: 'relative',
+      border: 'none',
+      outline: 'none',
+      padding: componentTheme.padding,
+      margin: '0',
+      width: '100%',
+      borderRadius: 'initial',
+      boxSizing: 'border-box',
+      textAlign: 'start',
+      cursor: 'pointer',
+      userSelect: 'none',
+      fontFamily: componentTheme.fontFamily,
+      fontWeight: componentTheme.fontWeight,
+      lineHeight: componentTheme.lineHeight,
+      fontSize: componentTheme.fontSize,
+      background: componentTheme.background,
+      transition: 'background 0.2s',
+      display: 'block',
```

Comment:
The original code had a comment about this line, I'd leave it here:

  Changing the following to display: flex; causes a VO
  bug where items with role menuitem, menuitemcheckbox
  and menuitemradio are selected twice with control+
  option+space. So we set the display to block.